src/agent/onefuzz/src/sanitizer.rs

```diff
@@ -2,18 +2,138 @@
 // Licensed under the MIT License.
 
 use std::collections::HashMap;
+use std::path::Path;
 
-use anyhow::Result;
+use anyhow::{bail, Result};
 
 pub fn default_llvm_symbolizer_path() -> Result<String> {
     Ok(std::env::var("LLVM_SYMBOLIZER_PATH")?)
 }
 
 pub fn default_sanitizer_env_vars() -> Result<HashMap<String, String>> {
-    let mut env = HashMap::default();
+    Ok(LlvmSymbolizer::from_env()?.sanitizer_env_vars())
+}
+
+/// Valid path to an `llvm-symbolizer` executable that can be used as an
+/// external symbolizer in sanitizer environment variables and options.
+#[derive(Clone, Debug)]
+pub struct LlvmSymbolizer {
+    path: String,
+}
+
+impl LlvmSymbolizer {
+    pub fn new(path: impl Into<String>) -> Result<Self> {
+        let path = path.into();
+
+        if !is_valid_symbolizer_path(&path) {
+            bail!("invalid symbolizer path: {}", path)
+        }
+
+        Ok(Self { path })
+    }
+
+    pub fn from_env() -> Result<Self> {
+        let path = default_llvm_symbolizer_path()?;
+
+        Self::new(path)
+    }
+
+    pub fn as_str(&self) -> &str {
+        &self.path
+    }
+
+    pub fn sanitizer_env_vars(&self) -> HashMap<String, String> {
+        let mut env = HashMap::default();
+
+        env.insert("ASAN_SYMBOLIZER_PATH".to_owned(), self.path.clone());
+
+        let options = format!("external_symbolizer={}", self.path);
```

Comment:
Wrong, should be `external_symbolizer_path` ([docs](https://github.com/google/sanitizers/wiki/ThreadSanitizerFlags)).